### PR TITLE
DVX-51 - Streamlit UI

### PR DESCRIPTION
**Pull request by jean-noumena**

### CODE REVIEW

File: streamlit-ui/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Set the working directory in the container to /app
+WORKDIR /app
+
+# Add the current directory contents into the container at /app
+ADD . /app
+
+# Install any needed packages specified in requirements.txt
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Make port 80 available to the world outside this container
```
Comment (ari-noumena) [inline streamlit-ui/Dockerfile:13]: That's not port 80!

I'd remove all the chatgpt comments 😛

File: streamlit-ui/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+./generated
+auth0-python
+pandas
+pyjwt
```
Comment (ari-noumena) [inline streamlit-ui/requirements.txt:4]: Protip: you probably want to pin/freeze (`pip freeze`) the versions of these!

Comment (ari-noumena) [reply streamlit-ui/requirements.txt:4]: How do I know? I didn't do it for `documentation` until things started breaking 🤣

File: streamlit-ui/src/app.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+
+import jwt
+import streamlit as st
+
+from src import iou
+from src import config
```
Comment (ari-noumena) [inline streamlit-ui/src/app.py:6]: For me (with venv) I need this to be

```suggestion
from python.src import iou
from python.src import config
```

But I'm not sure exactly how this works (yet)

File: streamlit-ui/src/app.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                access_token=st.session_state['access_token'],
+                host=config.ROOT_URL,
+                api_key={},
+                api_key_prefix={}
+            )
+        )
+    )
+    return api
+
+def createIou():
```
Comment (ari-noumena) [inline streamlit-ui/src/app.py:25]: Nit: Please use snake_case consistently :D

File: streamlit-ui/src/app.py (hunk 3 above)
Comment (jean-noumena) [reply streamlit-ui/src/app.py:6]: works in docker without the python. And why python?? I'm lost

File: streamlit-ui/src/app.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    import streamlit as st
```
Comment (ari-noumena) [inline streamlit-ui/src/app.py:26]: nit: I would put all import-related things at the top of the file, always

File: streamlit-ui/src/app.py (hunk 3 above)
Comment (ari-noumena) [reply streamlit-ui/src/app.py:6]: As discussed, this does not seem to be quite right, as it's certainly not the `python.src` thing we want to import 🤣

File: streamlit-ui/src/auth.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import requests
+from src import config
+from urllib.parse import urlencode
```
Comment (ari-noumena) [inline streamlit-ui/src/auth.py:3]: This looks to be unused

**Review by ari-noumena — DISMISSED** [review]
Next up: tests?! 😆

**Review by ari-noumena — APPROVED** [review]

**Pull request merged**